src/node/controller/messages.rs

```diff
@@ -184,4 +187,172 @@ impl Controller {
 
         Ok(())
     }
+
+    /// Returns the previous block if is a head block AND is a state_block
+    async fn previous_as_account_info(
+        &self,
+        previous_block_hash: &BlockHash,
+    ) -> anyhow::Result<Option<StateBlock>> {
+        let previous_block = Controller::block_by_hash(self, previous_block_hash).await?;
+        if let Some(previous_block) = previous_block {
+            let is_head = self
+                .get_latest_block(previous_block.account())
+                .await?
+                .map_or(false, |block| {
+                    let block_hash = block.hash().context("hash for is_head").unwrap();
+                    block_hash == previous_block_hash
+                });
+
+            dbg!(is_head);
```

Comment:
please remove, or use `trace!()` if useful to the developer or `debug!()` if useful to the user

Reply:
hehe oops, forgot this here